packages/wasm-dpp/src/identity/state_transition/identity_create_transition/identity_create_transition_basic_validator.rs

```diff
@@ -0,0 +1,153 @@
+            PublicKeysValidator<BlsAdapter>,
+            RequiredPurposeAndSecurityLevelValidator,
+            ExternalStateRepositoryLikeWrapper,
+            PublicKeysSignaturesValidator<BlsAdapter>,
+            BlsAdapter,
+        >,
+    ) -> Self {
+        Self(validator)
+    }
+}
+
+#[wasm_bindgen(js_class=IdentityCreateTransitionBasicValidator)]
+impl IdentityCreateTransitionBasicValidatorWasm {
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        state_repository: ExternalStateRepositoryLike,
+        js_bls: JsBlsAdapter,
+    ) -> Result<IdentityCreateTransitionBasicValidatorWasm, JsValue> {
+        let state_repository_wrapper =
+            Arc::new(ExternalStateRepositoryLikeWrapper::new(state_repository));
+
+        let protocol_version_validator = ProtocolVersionValidator::new(
+            LATEST_VERSION,
+            LATEST_VERSION,
+            COMPATIBILITY_MAP.clone(),
+        );
+
+        let public_keys_validator = PublicKeysValidator::new_with_schema(
+            PUBLIC_KEY_SCHEMA_FOR_TRANSITION.clone(),
+            BlsAdapter(js_bls.clone()),
+        )
+        .map_err(|e| from_dpp_err(ProtocolError::Generic(e.to_string())))?;
+
+        let public_keys_transition_validator = RequiredPurposeAndSecurityLevelValidator::default();
+
+        let asset_lock_proof_validator: AssetLockProofValidator<
+            ExternalStateRepositoryLikeWrapper,
+        >;
+        {
+            let tx_validator = Arc::new(AssetLockTransactionValidator::new(
+                state_repository_wrapper.clone(),
+            ));
+
+            let instant_asset_lock_proof_validator = InstantAssetLockProofStructureValidator::new(
+                state_repository_wrapper.clone(),
+                tx_validator.clone(),
+            )
+            .map_err(|e| from_dpp_err(ProtocolError::Generic(e.to_string())))?;
+
+            let chain_asset_lock_proof_validator =
+                ChainAssetLockProofStructureValidator::new(state_repository_wrapper, tx_validator)
+                    .map_err(|e| from_dpp_err(ProtocolError::Generic(e.to_string())))?;
+
+            asset_lock_proof_validator = AssetLockProofValidator::new(
+                instant_asset_lock_proof_validator,
+                chain_asset_lock_proof_validator,
+            );
+        }
+
+        let public_keys_signature_validator =
+            PublicKeysSignaturesValidator::new(BlsAdapter(js_bls.clone()));
+
+        let validator = IdentityCreateTransitionBasicValidator::new(
+            protocol_version_validator,
+            Arc::new(public_keys_validator),
+            Arc::new(public_keys_transition_validator),
+            Arc::new(asset_lock_proof_validator),
+            BlsAdapter(js_bls),
+            public_keys_signature_validator,
+        )
+        .map_err(|e| from_dpp_err(ProtocolError::Generic(e.to_string())))?;
+
+        Ok(validator.into())
+    }
+
+    #[wasm_bindgen(js_name=validate)]
```

Comment:
You don't need to write `js_name` if it's the same, just use `#[wasm_bindgen]`

Reply:
Fixed